src/common/utils/validateDuplicateLines.js

```diff
@@ -0,0 +1,38 @@
+import { batchFetch } from '@folio/stripes-acq-components';
+import { VALIDATION_ERRORS } from '../constants';
+
+export const validateDuplicateLines = (line, mutator) => {
+  const productIds = line.details?.productIds?.map(({ productId }) => productId) || [];
+  const baseQuery = `id<>"${line.id}" and titleOrPackage=="*${line.titleOrPackage}*"`;
+
+  return batchFetch(mutator.poLines, productIds, (itemsChunk) => {
+    const query = itemsChunk
+      .map(productId => `details.productIds=="*${productId}*"`)
+      .join(' and ');
+
+    return query
+      ? `${baseQuery} and (${query})`
+      : baseQuery;
+  }).then((existingLines) => {
```

Comment:
from what I see we follow style or async/await

```
promise
  .then
```

Reply:
We have different styles even in one function) Should I rewrite this function with async/await?